### PR TITLE
Improve the readme

### PR DESCRIPTION
**Pull request by PurpleBooth**
While this tool is used by core, the contributors who setup core and
maintain it are closely linked to homebrew project. As such they have in
depth knowledge of the domain and so are less likely to need this
document. There's also only one core (compared to the large number of
custom taps), as such this documentation may be helpful to core, but
it's not it's primary audience.

This document is written for the audience developers who don't have 100
stars (or whaver the required limit is) to get there formula into core,
and as such are creating a custom tap to distribute their application.

The key questions it aims to answer for those custom tap developers are:
* How do I test my new formula
* How to a automate the tests to that custom formula so people can
  contribute
* How do I build binaries for my formula so installs are fast

These are all tasks that test-bot significantly makes simpler, but
pretty much no one I talk to knows about it because these great features
aren't easily discoverable.

There are also a number of idiosyncrasies that happen on GitHub Actions
that need to be documented for a smooth "oh wow it just works"
experience for first time users.

This is also our opportunity to state to the user what we expect you to
do, and, by ommission, what we don't expect, hopefully reducing
questions like #474, #302, and #251

This should also live in the main man pages for the `test-bot` tool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,97 @@
-# Homebrew Test Bot
+Homebrew Test Bot
+=================
```
Comment (MikeMcQuaid) [inline README.md:2]: Please don't make changes like this, it makes the overall changes hard to follow based on your syntax preferences.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,97 @@
...
 
-Tests the full lifecycle of a Homebrew change to a tap (Git repository).
+Homebrew comes with a comprehensive suite of tools that lets you
+validate that both your homebrew installation is working and prove that
+a formula is working.
```
Comment (MikeMcQuaid) [inline README.md:6]:
```suggestion
Homebrew provides tools to verify your Homebrew installation and validate formula installation. Homebrew Test Bot uses these to automate the formula build and test process and upload the produced binaries.
```

File: README.md [hunk 3]
```diff
@@ -1,19 +1,97 @@
...
 
-## Install
+Homebrew Test Bot uses these tools to simplify the automation of your
+formula publishing process including Checking your installation,
+Checking your Formula, and Building and uploading binaries.
```
Comment (MikeMcQuaid) [inline README.md:10]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -1,19 +1,97 @@
...
 
 `brew test-bot` is automatically installed when first run.
 
-## Usage
+Usage
+-----
+
+Here are some of the most common use-cases for `brew test-bot`
+
+### Locally Validating a formula
```
Comment (MikeMcQuaid) [inline README.md:22]: I don't really consider this a common use-case.

File: README.md [hunk 5]
```diff
@@ -1,19 +1,97 @@
...
+    brew test-bot "custom/tap/your-formula"
+
+### Validating a formula change in GitHub actions
+
+You are running on GitHub actions test-bot will try to fix any potential
+problems in the homebrew repository deleting the `$(brew --repo)/Cellar`
+before performing any operations. If you've intentionally made changes
+there, they will be lost, what's more `--cleanup` is always passed.
+
+When `--cleanup` is passed, we hard reset the taps to "origin/master",
```
Comment (MikeMcQuaid) [inline README.md:44]: I'd rather we point to (and update, if necessary) `--help` instead of documenting behaviour here that may change in future.

File: README.md [hunk 6]
```diff
@@ -1,19 +1,97 @@
...
+as such your changes will need to be cloneable from "master".
+
+If you're using this tool for a personal tap, and as such won't have the
+changes on `master` because you're validating a pull request, or want to
+use another default branch, As a workaround, you can unset the
+`GITHUB_ACTIONS` environment variable to stop this behaviour, and
```
Comment (MikeMcQuaid) [inline README.md:50]: I don't want to recommend this here.

File: README.md [hunk 7]
```diff
@@ -1,19 +1,97 @@
...
+Then push the changes it's made to the tap to include the download
+information in the formula.
+
+    cd "$(brew --repo)/Library/Taps/custom/homebrew-tap"
+    git push
+
+### Full options
 
-See [the `brew test-bot` section of the `brew man` output](https://docs.brew.sh/Manpage#test-bot-options-formula) or `brew test-bot --help`.
+See [the `brew test-bot` section of the `brew man`
```
Comment (MikeMcQuaid) [inline README.md:83]: Please revert this.

File: README.md [hunk 8]
```diff
@@ -1,19 +1,97 @@
...
+output](https://docs.brew.sh/Manpage#test-bot-options-formula) or
+`brew test-bot --help`.
 
-## Tests
+Tests
+-----
 
-Tests can be run with `bundle install && bundle exec rspec`.
+You can run the tests with `bundle install && bundle exec rspec`.
```
Comment (MikeMcQuaid) [inline README.md:90]: Please revert this.

**Comment by MikeMcQuaid** [comment]
> This should also live in the main man pages for the `test-bot` tool.

If so: they should not be duplicated here and should instead be in https://github.com/Homebrew/homebrew-test-bot/blob/master/cmd/test-bot.rb.



> This document is written for the audience developers who don't have 100
> stars (or whaver the required limit is) to get there formula into core,
> and as such are creating a custom tap to distribute their application.

They should be using `brew tap-new` to setup their custom tap which will configure `brew test-bot` for them (and remove most of the need for the manual steps here). Can we point to that instead?

File: README.md (hunk 6 above)
Comment (PurpleBooth) [reply README.md:50]: I think I'll swap out this whole section to just a be a reference to the `tap-new` command
